assets/js/ajax-filters.js

```diff
@@ -34,14 +34,17 @@ jQuery( document ).ready( function( $ ) {
 	 * Get the session storage key for the job listings instance.
 	 */
 	function get_session_storage_key( $target ) {
-		var index          = $( 'div.job_listings' ).index( $target );
-		var unique_page_id = $target.data( 'post_id' );
+		var index   = $( 'div.job_listings' ).index( $target );
+		var pageUrl = window.location.href;
```

Comment:
I'm not sure what this is solving as we were already falling back to the URL if the page ID wasn't being set. I'd prefer to use page ID where possible and fall back to URL. I believe we added that fallback for themes like yours which  have `[jobs]` elements on non-`[jobs]` pages.

Reply:
Yep, I understand the reasoning behind it and the history of the commits, but as I said in #1891 the behavior is not consistent when the form is submitting filter values from one page to another, especially when on one page is with the fallback URL in the key name and on the redirected page is with page ID.

Reply:
Shouldn't it only persist on the same page? I'm not sure I understand the steps to reproduce still.